test/db.go

```diff
@@ -62,7 +63,8 @@ func resetTestDatabase(t testing.TB, dbPrefix string) func() {
 // counters. See allTableNamesInDB for what is meant by "all tables
 // available". To be used only in test code.
 func deleteEverythingInAllTables(db CleanUpDB) error {
-	ts, err := allTableNamesInDB(db)
+	ctx := context.Background()
```

Comment:
a) why does this not take a context, but `allTableNamesInDB` does?

then, probably non-actionable in this PR:
b) why does neither of these take a `t` and mark itself as `t.Helper()`?
c) gosh I wish that `t` came with a `t.Context()` that would give you a handy dandy context with the test's metadata and timeout (if any) already populated.

Reply:
I wound up plumbing up the context as far as the two exported functions, ResetBoulderTestDatabase and ResetIncidentsTestDatabase. The former is called a _lot_ of places. I don't really know of a principled approach to when we should pass through a context in test code or not. In general we're not actually applying timeouts or doing traces. That would argue for never passing a context and always using `context.Background()` all the way down at the leaf functions. But that feels... bad.